base/common/src/main/java/org/dogtagpki/acme/ACMEIdentifier.java

```diff
@@ -44,6 +44,92 @@ public void setValue(String value) {
         this.value = value;
     }
 
+    /**
+     * Validate syntax of identifier.
+     *
+     * @throws IllegalArgumentException if either type or value is null.
+     * @return a result that is either OK, or a failure with an
+     *         appropriate ACMEError attached.
+     */
+    public ValidationResult validateSyntax() {
```

Comment:
My issue with this is that the `ACMEIdentifier` class is supposed to represent the identifier in ACMEv2 protocol and can be shared between server & client. However, now this class will be dependent on `ValidationResult` which is not part of ACMEv2 protocol and probably should remain on the server side only.

Reply:
OK, I'll break the validation behaviour into a separate class in the `server` subpackage.